extractor/src/test/java/org/schabi/newpipe/extractor/services/youtube/YoutubeDashManifestCreatorTest.java

```diff
@@ -0,0 +1,620 @@
+            assertNotNull(mediaPresentationDurationValue,
+                    "The value of the mediaPresentationDuration attribute is empty or the corresponding attribute doesn't exist");
+            assertTrue(mediaPresentationDurationValue.startsWith("PT"),
+                    "The mediaPresentationDuration attribute of the DASH manifest is not valid");
+        }
+
+        private void testPeriodElement(@Nonnull final Document document) {
+            final Element periodElement = (Element) document.getElementsByTagName("Period")
+                    .item(0);
+            assertNotNull(periodElement, "The Period element doesn't exist");
+            assertTrue(periodElement.getParentNode().isEqualNode(
+                            document.getElementsByTagName("MPD").item(0)),
+                    "The MPD element doesn't contain a Period element");
+        }
+
+        private void testAdaptationSetElement(@Nonnull final Document document,
+                                              @Nonnull final ItagItem itagItem) {
+            final Element adaptationSetElement = (Element) document
+                    .getElementsByTagName("AdaptationSet").item(0);
+            assertNotNull(adaptationSetElement, "The AdaptationSet element doesn't exist");
+            assertTrue(adaptationSetElement.getParentNode().isEqualNode(
+                            document.getElementsByTagName("Period").item(0)),
+                    "The Period element doesn't contain an AdaptationSet element");
+
+            final String mimeTypeDashManifestValue = adaptationSetElement
+                    .getAttribute("mimeType");
+            assertFalse(isBlank(mimeTypeDashManifestValue),
+                    "The value of the mimeType attribute is empty or the corresponding attribute doesn't exist");
+
+            final String mimeTypeItagItemValue = itagItem.getMediaFormat().getMimeType();
+            assertFalse(isBlank(mimeTypeItagItemValue), "The mimeType of the ItagItem is empty");
+
+            assertEquals(mimeTypeDashManifestValue, mimeTypeItagItemValue,
+                    "The mimeType attribute of the DASH manifest (" + mimeTypeItagItemValue
+                            + ") is not equal to the mimeType set in the ItagItem object ("
+                            + mimeTypeItagItemValue + ")");
+        }
+
+        private void testRoleElement(@Nonnull final Document document) {
+            final Element roleElement = (Element) document.getElementsByTagName("Role")
+                    .item(0);
+            assertNotNull(roleElement, "The Role element doesn't exist");
+            assertTrue(roleElement.getParentNode().isEqualNode(
+                            document.getElementsByTagName("AdaptationSet").item(0)),
+                    "The AdaptationSet element doesn't contain a Role element");
+        }
+
+        private void testRepresentationElement(@Nonnull final Document document,
```

Comment:
Yes, these tests could be improved by splitting them or using `assertAll`. If you want I can take care of this @TiA4f8R